trending_data/src/main/webapp/unique_window_handler.js

```diff
@@ -144,17 +148,42 @@ export class UniqueWindowHandler {
     }
     this.lastCode = this.marker.countryCode;
     this.initDataWindow();
-    const ytContent = await this.loadYTData();
-    this.loadTwitterData();
+    await this.loadYTCategories();
+    await this.loadYTData(this.defaultYTCategory);
+    await this.loadTwitterData();
```

Comment:
The asynchronous nature is somewhat wasted here, since these will executed/waited for in sequence.
One solution is to use Promise.all, but this would mean that any rejection (from one of the three) would cause the whole promise to be rejected. There are ways around this.

This stackoverflow post has some answers/details: https://stackoverflow.com/questions/35612428/call-async-await-functions-in-parallel

You don't have to worry about this now, but we should maybe think about adding a kanban task to address this. It would almost certainly make the pop-ups load quicker.